File: 6-paradigmas/1932.cpp
#include<iostream>
//
int main(){
	int quotes,price;
	scanf("%i %i",&quotes,&price);
	int left,right,tot_temp=0,tot=0;
	scanf("%i",&left);
	while(quotes>1){
		scanf("%i",&right);
		if(right-left>=tot_temp){
			tot_temp=std::max(tot_temp,right-left-price);
		}else if(tot_temp){
			tot+=tot_temp;
			left=right;
			tot_temp=0;
		}else{
			left=std::min(left,right);
		}
		--quotes;
	}
	tot+=tot_temp;
	printf("%i\n",tot);

	return 0;
}

File: 6-paradigmas/1932.php
<?php
list($quotes,$price)=explode(' ',readLine());
$price=(int)$price;

$left=10001;
$tot_temp=0;
$tot=0;
foreach(explode(' ',readLine()) as $right){
	if($right-$left>=$tot_temp){
		$tot_temp=($right-$left-$price>$tot_temp)?$right-$left-$price:$tot_temp;
	}else if($tot_temp){
		$tot+=$tot_temp;
		$tot_temp=0;
		$left=$right;
	}else{
		$left=($right<$left)?$right:$left;
	}
}
$tot+=$tot_temp;
print $tot;
print "\n";

?>
